src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/Link.js

```diff
@@ -0,0 +1,226 @@
+// @flow
+import React, {Component, Fragment} from 'react';
+import classNames from 'classnames';
+import {observer} from 'mobx-react';
+import {action, observable} from 'mobx';
+import SingleSelect from '../../components/SingleSelect/SingleSelect';
+import Icon from '../../components/Icon';
+import linkStyles from '../Form/fields/link.scss';
+import linkTypeRegistry from './registries/linkTypeRegistry';
+import type {ChangeContext} from '../Form/types';
+import type {LinkTypeValue} from './types';
+
+type Props = {
+    disabled?: boolean,
+    enableAnchor?: ?boolean,
+    enableTarget?: ?boolean,
+    locale: string,
+    onChange: (value: LinkTypeValue, context?: ChangeContext) => void,
+    onFinish: (subDataPath: ?string, subSchemaPath: ?string) => void,
+    types?: string[],
+    value: ?LinkTypeValue,
+}
+
+const DEFAULT_TARGET = '_self';
+
+@observer
+class Link extends Component<Props> {
+    static defaultProps = {
+        disabled: false,
+        enableAnchor: false,
+        enableTarget: false,
+        types: [],
+    };
+
+    @observable provider: ?string;
+    @observable openOverlay: ?string;
+    @observable target: ?string;
+    @observable href: ?string | number;
+    @observable title: ?string = '';
+    @observable anchor: ?string;
+
+    @observable overlayHref: ?string | number;
+    @observable overlayProvider: ?string;
+    @observable overlayTitle: ?string;
+    @observable overlayTarget: ?string = DEFAULT_TARGET;
+    @observable overlayAnchor: ?string;
+
+    constructor(props: Props): void {
+        super(props);
+        const {value} = this.props;
+
+        if (value) {
+            const {provider, title, href, target, anchor} = value;
+            this.provider = provider;
```

Comment:
i think you could always read these values from `this.props.value` instead of storing them in the component? or do you need to change them without setting the change to the form somewhere?